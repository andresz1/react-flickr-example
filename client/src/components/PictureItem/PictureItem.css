.container {
  min-width: 280px;
  max-width: 280px;
  margin: 8px;
}

.header {
  padding: 8px 0;
}

.title,
.description {
  width: 100%;
  text-align: right;
  font-size: 12px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.title {
  color: #0063DC;
}
