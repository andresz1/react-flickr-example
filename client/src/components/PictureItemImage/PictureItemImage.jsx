// @flow

import * as React from 'react';
import styles from './PictureItemImage.css';

type IProps = {
  children?: React.Node,
  src?: string,
  onClick: Function,
};

const PictureItemImage = (props: IProps) => (
  <div
    className={styles.container}
    style={{backgroundImage: (props.src)? `url(${props.src})`: ''}}
    onClick={props.onClick}
  >
    {props.children}
  </div>
);

export default PictureItemImage;
