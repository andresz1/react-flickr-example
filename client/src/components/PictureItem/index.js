import PictureItem from './PictureItem.jsx';

export default PictureItem;
