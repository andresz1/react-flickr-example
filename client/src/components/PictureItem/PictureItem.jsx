// @flow

import React from 'react';
import ImageLoader from 'react-load-image';
import { BubbleLoader } from 'react-css-loaders';

import PictureItemImage from '../PictureItemImage';
import styles from './PictureItem.css';

type IProps = {
  picture: IPicture,
  onSelect: Function
};

const PictureItem = (props: IProps) => {
  const { picture } = props;

  return(
    <div className={styles.container}>
      <div className={styles.header}>
        <div className={styles.title}>
          {props.picture.title ||Â '-'}
        </div>

        <div className={styles.description}>by&nbsp;
          <a
            href={picture.href}
            target="_blank">
            {picture.ownername}
          </a>
        </div>
      </div>

      <ImageLoader src={picture.src}>
        <PictureItemImage onClick={() => props.onSelect(props.picture)}/>

        <PictureItemImage>
          Error
        </PictureItemImage>

        <PictureItemImage>
          <BubbleLoader
            color="#fff"
            size="4"
          />
        </PictureItemImage>
      </ImageLoader>
    </div>
  );
};

export default PictureItem;
