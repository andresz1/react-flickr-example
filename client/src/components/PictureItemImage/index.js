import PictureItemImage from './PictureItemImage.jsx';

export default PictureItemImage;
