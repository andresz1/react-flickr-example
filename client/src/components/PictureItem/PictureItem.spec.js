import React from 'react';
import { shallow } from 'enzyme';
import PictureItem from './PictureItem';
import PictureItemImage from '../PictureItemImage';

const picture = {
  id: '44592167171',
  owner: '166588675@N05',
  secret: '144e4f8561',
  server: '1848',
  farm: 2,
  title: 'Olden 3',
  ispublic: 1,
  isfriend: 0,
  isfamily: 0,
  ownername: 'OldenCam3',
  tags: '',
  src: 'https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg',
  href: 'https://www.flickr.com/photos/166588675@N05/44592167171/'
};

const setup = (setupProps = {}) => {
  const defaultProps = {
    picture,
    handleSelect: jest.fn()
  };
  const props = { ...defaultProps, ...setupProps };
  const wrapper = shallow(
    <PictureItem
      picture={props.picture}
      onSelect={props.handleSelect}
    />
  );

  return {
    props,
    wrapper
  };
};

describe('<PictureItem />', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('puts a dash when there is no title', () => {
    const { wrapper } = setup({ picture: {
      ...picture,
      title: undefined,
    }});
    expect(wrapper).toMatchSnapshot();
  });

  it('calls onSelect when image clicked', () => {
    const { props, wrapper } = setup();

    wrapper.find(PictureItemImage).first().simulate('click');
    expect(props.handleSelect).toHaveBeenCalled();
  });
});
