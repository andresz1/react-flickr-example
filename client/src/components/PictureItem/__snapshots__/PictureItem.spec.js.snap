// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PictureItem /> puts a dash when there is no title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PictureItem
    onSelect={[Function]}
    picture={
        Object {
            "farm": 2,
            "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
            "id": "44592167171",
            "isfamily": 0,
            "isfriend": 0,
            "ispublic": 1,
            "owner": "166588675@N05",
            "ownername": "OldenCam3",
            "secret": "144e4f8561",
            "server": "1848",
            "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
            "tags": "",
            "title": undefined,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <div
                    className="title"
          >
                    -
          </div>
          <div
                    className="description"
          >
                    by 
                    <a
                              href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                              target="_blank"
                    >
                              OldenCam3
                    </a>
          </div>
</div>,
        <ImageLoader
          src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
>
          <PictureItemImage
                    onClick={[Function]}
          />
          <PictureItemImage>
                    Error
          </PictureItemImage>
          <PictureItemImage>
                    <m
                              color="#fff"
                              duration={1.8}
                              size="4"
                    />
          </PictureItemImage>
</ImageLoader>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              -
</div>,
            <div
              className="description"
>
              by 
              <a
                            href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                            target="_blank"
              >
                            OldenCam3
              </a>
</div>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "-",
              "className": "title",
            },
            "ref": null,
            "rendered": "-",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "by ",
                <a
                  href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                  target="_blank"
>
                  OldenCam3
</a>,
              ],
              "className": "description",
            },
            "ref": null,
            "rendered": Array [
              "by ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "OldenCam3",
                  "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "OldenCam3",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PictureItemImage
              onClick={[Function]}
/>,
            <PictureItemImage>
              Error
</PictureItemImage>,
            <PictureItemImage>
              <m
                            color="#fff"
                            duration={1.8}
                            size="4"
              />
</PictureItemImage>,
          ],
          "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Error",
            },
            "ref": null,
            "rendered": "Error",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <m
                color="#fff"
                duration={1.8}
                size="4"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#fff",
                "duration": 1.8,
                "size": "4",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <div
                        className="title"
            >
                        -
            </div>
            <div
                        className="description"
            >
                        by 
                        <a
                                    href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                                    target="_blank"
                        >
                                    OldenCam3
                        </a>
            </div>
</div>,
          <ImageLoader
            src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
>
            <PictureItemImage
                        onClick={[Function]}
            />
            <PictureItemImage>
                        Error
            </PictureItemImage>
            <PictureItemImage>
                        <m
                                    color="#fff"
                                    duration={1.8}
                                    size="4"
                        />
            </PictureItemImage>
</ImageLoader>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                -
</div>,
              <div
                className="description"
>
                by 
                <a
                                href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                                target="_blank"
                >
                                OldenCam3
                </a>
</div>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "-",
                "className": "title",
              },
              "ref": null,
              "rendered": "-",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "by ",
                  <a
                    href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                    target="_blank"
>
                    OldenCam3
</a>,
                ],
                "className": "description",
              },
              "ref": null,
              "rendered": Array [
                "by ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "OldenCam3",
                    "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "OldenCam3",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PictureItemImage
                onClick={[Function]}
/>,
              <PictureItemImage>
                Error
</PictureItemImage>,
              <PictureItemImage>
                <m
                                color="#fff"
                                duration={1.8}
                                size="4"
                />
</PictureItemImage>,
            ],
            "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Error",
              },
              "ref": null,
              "rendered": "Error",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <m
                  color="#fff"
                  duration={1.8}
                  size="4"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "#fff",
                  "duration": 1.8,
                  "size": "4",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`<PictureItem /> renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PictureItem
    onSelect={[Function]}
    picture={
        Object {
            "farm": 2,
            "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
            "id": "44592167171",
            "isfamily": 0,
            "isfriend": 0,
            "ispublic": 1,
            "owner": "166588675@N05",
            "ownername": "OldenCam3",
            "secret": "144e4f8561",
            "server": "1848",
            "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
            "tags": "",
            "title": "Olden 3",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header"
>
          <div
                    className="title"
          >
                    Olden 3
          </div>
          <div
                    className="description"
          >
                    by 
                    <a
                              href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                              target="_blank"
                    >
                              OldenCam3
                    </a>
          </div>
</div>,
        <ImageLoader
          src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
>
          <PictureItemImage
                    onClick={[Function]}
          />
          <PictureItemImage>
                    Error
          </PictureItemImage>
          <PictureItemImage>
                    <m
                              color="#fff"
                              duration={1.8}
                              size="4"
                    />
          </PictureItemImage>
</ImageLoader>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              Olden 3
</div>,
            <div
              className="description"
>
              by 
              <a
                            href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                            target="_blank"
              >
                            OldenCam3
              </a>
</div>,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Olden 3",
              "className": "title",
            },
            "ref": null,
            "rendered": "Olden 3",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "by ",
                <a
                  href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                  target="_blank"
>
                  OldenCam3
</a>,
              ],
              "className": "description",
            },
            "ref": null,
            "rendered": Array [
              "by ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "OldenCam3",
                  "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "OldenCam3",
                "type": "a",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PictureItemImage
              onClick={[Function]}
/>,
            <PictureItemImage>
              Error
</PictureItemImage>,
            <PictureItemImage>
              <m
                            color="#fff"
                            duration={1.8}
                            size="4"
              />
</PictureItemImage>,
          ],
          "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Error",
            },
            "ref": null,
            "rendered": "Error",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <m
                color="#fff"
                duration={1.8}
                size="4"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "#fff",
                "duration": 1.8,
                "size": "4",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header"
>
            <div
                        className="title"
            >
                        Olden 3
            </div>
            <div
                        className="description"
            >
                        by 
                        <a
                                    href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                                    target="_blank"
                        >
                                    OldenCam3
                        </a>
            </div>
</div>,
          <ImageLoader
            src="https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg"
>
            <PictureItemImage
                        onClick={[Function]}
            />
            <PictureItemImage>
                        Error
            </PictureItemImage>
            <PictureItemImage>
                        <m
                                    color="#fff"
                                    duration={1.8}
                                    size="4"
                        />
            </PictureItemImage>
</ImageLoader>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                Olden 3
</div>,
              <div
                className="description"
>
                by 
                <a
                                href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                                target="_blank"
                >
                                OldenCam3
                </a>
</div>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Olden 3",
                "className": "title",
              },
              "ref": null,
              "rendered": "Olden 3",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "by ",
                  <a
                    href="https://www.flickr.com/photos/166588675@N05/44592167171/"
                    target="_blank"
>
                    OldenCam3
</a>,
                ],
                "className": "description",
              },
              "ref": null,
              "rendered": Array [
                "by ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "OldenCam3",
                    "href": "https://www.flickr.com/photos/166588675@N05/44592167171/",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "OldenCam3",
                  "type": "a",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PictureItemImage
                onClick={[Function]}
/>,
              <PictureItemImage>
                Error
</PictureItemImage>,
              <PictureItemImage>
                <m
                                color="#fff"
                                duration={1.8}
                                size="4"
                />
</PictureItemImage>,
            ],
            "src": "https://farm2.staticflickr.com/1848/44592167171_144e4f8561.jpg",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Error",
              },
              "ref": null,
              "rendered": "Error",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <m
                  color="#fff"
                  duration={1.8}
                  size="4"
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "#fff",
                  "duration": 1.8,
                  "size": "4",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
